doc/architecture-decisions/0016-job-for-upgrading-istio-sidecars-on-workloads.md

```diff
@@ -0,0 +1,55 @@
+# 16. Job for Upgrading Istio Sidecars on Workloads
+
+Date: 2020-08-11
+
+## Status
+
+Accepted
+
+## Context
+
+Istio's service mesh capabilites are facilitated via sidecars injected into
+workload pods. These sidecars run an Istio-patched version of Envoy that is tied
+to the version of Istio that injects them.
+
+Typically when new versions of Istio are released, new versions of the sidecars
+are released as well. Istio has been good so far about supporting older versions
+of sidecars that were deployed before Istio was upgraded, but it is still
+[documented best practice](https://istio.io/latest/docs/setup/upgrade/) to roll
+all the pods after an Istio upgrade.
+
+As an additional constraint, the operators of cf-for-k8s clusters expect to be
+able to perform upgrades in one `kapp deploy`, with no post-install hooks or
+other bash scripts. This limits our options considerably. See this [Slack
+thread](https://cloudfoundry.slack.com/archives/CH9LF6V1P/p1592521879117400) on
+that constraint.
+
+## Decision
+
+We will use the kubernetes
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) resource
+to run the kubectl command needed to roll workload pods, after waiting for the
+new Istio control plane to be up and healthy.
+
+To that end, we will add the necessary minimal `ServiceAccounts` and `Roles`
+needed to list resources in the `istio-system` namespace, and restart resources
```

Comment:
The mechanism described here should serve our needs for now, so I'm not suggesting any changes here.

But there are other reasons to restart the app pods as well (such as whatever the equivalent of a rootfs rollout is these days). I wonder if one day in the future we can all unify on a mechanism to trigger these.

Somethin' to chew on!!!! 🍽

cc @cwlbraa

Reply:
Today in capiland, we're modifying the OPI-templated StatefulSet spec to trigger controlled rootfs rollouts.